internal/pkg/admission/policy-server-deployment.go

```diff
@@ -118,9 +118,31 @@ func (r *Reconciler) updatePolicyServerDeployment(ctx context.Context, policySer
 	return nil
 }
 
-func shouldUpdatePolicyServerDeployment(originalDeployment *appsv1.Deployment, newDeployment *appsv1.Deployment) bool {
+func getPolicyServerImageFromDeployment(policyServer *v1alpha2.PolicyServer, deployment *appsv1.Deployment) (string, error) {
+	for containerIndex := range deployment.Spec.Template.Spec.Containers {
+		container := &deployment.Spec.Template.Spec.Containers[containerIndex]
+		if container.Name == policyServer.NameWithPrefix() {
+			return container.Image, nil
+		}
+	}
+	return "", fmt.Errorf("cannot find policy server container")
+}
+
+func ifPolicyServerImageChanged(policyServer *v1alpha2.PolicyServer, originalDeployment *appsv1.Deployment, newDeployment *appsv1.Deployment) bool {
+	var oldImage, newImage string
+	var err error
+	if oldImage, err = getPolicyServerImageFromDeployment(policyServer, originalDeployment); err != nil {
+		return false
+	}
+	if newImage, err = getPolicyServerImageFromDeployment(policyServer, newDeployment); err != nil {
+		return false
+	}
```

Comment:
I would not hide these errors. Instead, I would:

  * Change the signature of the function to return `(string, error)`
  * Change these `return` to be `return "", false`

Reply:
I've changed to code to propagate the error. Thus, I believe that  reconciler can handle it better. Please, let me know what you think.